### PR TITLE
docs(headless): add PR reference docs

### PR DESCRIPTION
**Pull request by spiracha-coveo**
### Description:
- Added reference documentation for the PR controller.
- The structure of these controllers has been changed to follow a similar structure to what is currently present it the `headless-cart-recommendations.ts` file. This was done so that I could write string annotations describing the controller's properties and methods. 

### Questions
1. I tested the changes I made to various types by writing a test which checked the controller object's shape before and after my modification to ensure it remained the same. I have left these tests in for now; however, I will remove them before merging this PR. Please let me know if there is a better way to test my changes though.
2. `CartRecommendationsListState` is the only state object for all these controllers that uses the `ErrorPayload` type for the corresponding `error` property. The rest use the `SearchAPIErrorWithStatusCode` as defined by the base `ProductRecommendationsList` controller. I just wanted to confirm if this is expected.
3. For `FrequentlyBoughtTogetherList`, the `optionsSchema` is modified so that only a single `sku` is _required_. This means that one `sku` must be provided on initialization of this controller. However, if you looked at `FrequentlyBoughtTogetherListProps`, it noted that `options` was optional. I have changed this to mandatory, let me know if this is correct.

[LENS-665]: https://coveord.atlassian.net/browse/LENS-665?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/controllers/product-recommendations/headless-frequently-viewed-together.test.ts [hunk 1]
```diff
@@ -28,4 +28,14 @@ describe('headless frequently-viewed-together', () => {
...
+  it('object shape matches original', () => {
+    expect(frequentlyViewedTogether.refresh).toBeTruthy();
+    expect(frequentlyViewedTogether.setSkus).toBeTruthy();
+    expect(frequentlyViewedTogether.subscribe).toBeTruthy();
+    expect(frequentlyViewedTogether.state.error).toBeFalsy();
+    expect(frequentlyViewedTogether.state.isLoading).toBeFalsy();
+    expect(frequentlyViewedTogether.state.maxNumberOfRecommendations).toBe(5);
+    expect(frequentlyViewedTogether.state.recommendations).toBeTruthy();
+    expect(frequentlyViewedTogether.state.skus).toBeTruthy();
+  });
```
Comment (louis-bompart) [inline packages/headless/src/controllers/product-recommendations/headless-frequently-viewed-together.test.ts:40]: LGTM, tho in the future @coveo/search, maybe we should think about using Zod or Bueno to better transcribe the obj structure?

**Review by jpmarceau — APPROVED** [review]
Nice!

**Review by samisayegh — APPROVED** [review]
Great effort, thank you! 🙏

File: packages/headless/src/controllers/product-recommendations/headless-frequently-bought-together.ts [hunk 2]
```diff
@@ -13,24 +16,89 @@ const optionsSchema = new Schema({
...
+   */
+  maxNumberOfRecommendations: number;
+
+  /**
+   * The products recommended by the Coveo platform.
+   */
+  recommendations: ProductRecommendation[];
+
+  /**
+   * An error returned by the Coveo platform when executing a recommendation request, if any. This is `null` otherwise.
```
[the remark this answers is not in the document]
Comment (spiracha-coveo) [reply packages/headless/src/controllers/product-recommendations/headless-frequently-bought-together.ts]: Done!

File: packages/headless/src/controllers/product-recommendations/headless-frequently-bought-together.ts [hunk 3]
```diff
@@ -13,24 +16,89 @@ const optionsSchema = new Schema({
...
 });
 
-export type FrequentlyBoughtTogetherListOptions = SchemaValues<
-  typeof optionsSchema
->;
+export interface FrequentlyBoughtTogetherListOptions {
+  /**
+   * The SKU of the product to fetch recommendations for.
+   */
+  sku: string;
```
[the remark this answers is not in the document]
Comment (spiracha-coveo) [reply packages/headless/src/controllers/product-recommendations/headless-frequently-bought-together.ts:23]: Discussed in person - my change is valid

File: packages/headless/src/controllers/product-recommendations/headless-frequently-bought-together.ts [hunk 4]
```diff
@@ -13,24 +16,89 @@ const optionsSchema = new Schema({
...
+
+  /**
+   * Additional fields to fetch in the results.
+   */
+  additionalFields?: string[] | null | undefined;
+}
 
 export interface FrequentlyBoughtTogetherListProps {
-  options?: FrequentlyBoughtTogetherListOptions;
+  options: FrequentlyBoughtTogetherListOptions;
```
[the remark this answers is not in the document]
Comment (spiracha-coveo) [reply packages/headless/src/controllers/product-recommendations/headless-frequently-bought-together.ts:39]: Discussed in person - my change is valid

**Pull request merged**